cli/lib/kontena/cli/stacks/yaml/reader.rb

```diff
@@ -83,14 +101,20 @@ def execute(service_name = nil)
       end
 
       def stack_name
-        yaml = ::YAML.safe_load(raw_content)
-        yaml['stack'].split('/').last.split(':').first if yaml['stack']
+        yaml = ::YAML.safe_load(yaml_section('stack').to_s)
+        if yaml
+          yaml['stack'].split('/').last.split(':').first if yaml['stack']
+        end
       end
 
       def stack_version
         yaml['version'] || yaml['stack'].to_s[/:(.*)/, 1] || '1'
       end
 
+      def yaml_section(section)
+        raw_content[/^(#{section}:.+?)^\S/m, 1]
```

Comment:
How safe is this? We should add more tests for this method If we really need this kind of parsing.